### PR TITLE
Use turn tag to prevent suppression of designated turns, #3365

### LINKED ISSUE #3365: Obvious Turns Fail to Recognise Turning Roads in segregated Intersections
Our current obvious turn discovery fails in cases like this:
screen shot 2016-11-24 at 16 41 24

Here the road keeps the same reference, but follows a series of different roads.

I've added a test over at https://github.com/Project-OSRM/osrm-backend/pull/3364. 

Currently, the turn to the left at the intersection is considered obvious, the one to the right is announced as a turn.

In my opinion, the turn to the left and the turn to the right should both not be classified obvious.

/cc @karenzshea

### PR DESCRIPTION
**Pull request by oxidase**
# Issue

Fixes #3365 by using [`turn`](https://wiki.openstreetmap.org/wiki/Key:turn?uselang=en-US) tag.

At the moment the new flag `turn` is used only to prevent suppression of `Continue` instruction if ways have tags `turn=left` or `turn=right`

/cc @MoKob 

## Tasklist
 - [ ] add regression / cucumber cases (see docs/testing.md)
 - [ ] check places with modified instructions
 - [ ] review
 - [ ] adjust for comments

## Requirements / Relations
 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: profiles/lib/way_handlers.lua [hunk 1]
```diff
@@ -82,6 +82,14 @@ function WayHandlers.startpoint(profile,way,result,data)
   end
 end
 
+-- determine if this way has the indicated direction in which a way will lead
+function WayHandlers.turn(profile,way,result,data)
+  local turn = way:get_value_by_key("turn")
+  if turn == "left" or turn == "right" then
+    result.is_turn = true
+  end
+end
```
Comment (daniel-j-h) [inline profiles/lib/way_handlers.lua]: Needs an update in the taginfo.json (and maybe also the changelog)

**Comment by daniel-j-h** [comment]
Hm from the osm wiki the `turn=*` tag is not clear to me - `turn:lanes=*` are for turn lanes, the `turn=*` tag is for a single ways indicated direction? I can also only see a few thousand of these in the wild:
 
https://taginfo.openstreetmap.org/keys/turn#values

Could you clarify the use-case here in this scenario where turn lanes are tagged initially:

https://www.openstreetmap.org/edit#map=20/49.48756/8.47643

**Comment by MoKob** [comment]
@oxidase I don't feel that using barely supported tags is the way to go here. It feels like the *easy way out*. Are there other adjustments we could consider to prevent the suppression of an instruction? I feel that #4273 would be a better way to detect this and properly collapse it.

**Comment by TheMarex** [comment]
Hrm I've looked at the usages around Germany and this tags seems primarily used for tagging sliproads. Maybe this would be a nice addition for the sliproad handler to augment automatic detection?

**Comment by oxidase** [comment]
@TheMarex i agree here with @MoKob as we need to fix #4273 first and recheck the particular intersection. I  think it can be closed for now and it is always possible to restore the branch if it will be useful in the future.

**Pull request closed without merging**